### PR TITLE
task/improve grant creator request emails

### LINKED ISSUE #207: Update outgoing approved Grant Request email
Change to:

Thank you for logging into Competitions! 

After you log in, you can create a Grant by visiting:
https://competitions.cd2h.org/profile/grants

If you have any questions or would like to request a demo, please reply to this email. 

Happy grantmaking!

### PR DESCRIPTION
**Pull request by jseraf**
Closes #207 

- Update mailer text
- Add GrantCreatorRequest factory
- GrantCreatorRequest model spec
- GrantCreatorRequestReviewMailer spec

### CODE REVIEW

File: spec/models/grant_creator_request_spec.rb [hunk 1]
```diff
@@ -1,5 +1,93 @@
...
+      expect(request.errors).to include(:request_comment)
+    end
+
+    it 'disallows request from system_admin' do
+      request.requester.update_attribute(:system_admin, true)
+      expect(request).not_to be_valid
+      expect(request.errors[:base]).to include I18n.t('activerecord.errors.models.grant_creator_request.attributes.base.is_system_admin')
+    end
+
+    it 'disallows request from system_admin' do
```
Comment (mattbaumann1) [inline spec/models/grant_creator_request_spec.rb]: Should this label be someone who already has privilege.

Comment (jseraf) [reply spec/models/grant_creator_request_spec.rb]: yes.

File: spec/factories/grant_creator_request.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    trait :reviewed_by_system_admin do
+      association :reviewer, factory: :system_admin_user
+    end
+
+    factory :approved_grant_creator_request, traits: %i[approved]
+    factory :rejected_grant_creator_request, traits: %i[rejected]
+
+    factory :reviewed_approved_grant_creator_request, traits: %i[approved reviewed_by_system_admin]
+    factory :rejected_rejected_grant_creator_request, traits: %i[rejected reviewed_by_system_admin]
```
Comment (fanglinnw) [inline spec/factories/grant_creator_request.rb]: why two rejected_?

File: spec/models/grant_creator_request_spec.rb [hunk 3]
```diff
@@ -1,5 +1,93 @@
...
+      expect(request.errors[:base]).to include I18n.t('activerecord.errors.models.grant_creator_request.attributes.base.has_grant_creator_access')
+    end
+
+    it 'disallows request from user with an exisiting pending request' do
+      request.save
+      expect(FactoryBot.build(:grant_creator_request, requester: request.requester)).not_to be_valid
+    end
+
+    it 'allows request from user with an exisiting rejected request' do
+      request.update_attribute(:status, 'Approved')
```
Comment (fanglinnw) [inline spec/models/grant_creator_request_spec.rb]: not sure if the description matches this test

File: app/models/grant_creator_request.rb [hunk 4]
```diff
@@ -45,17 +45,15 @@ def requester_is_not_a_system_admin
...
+    errors.add(:base, :reviewer_is_not_system_admin) unless reviewer.system_admin?
   end
 
-  def udpate_user_grant_creator_attribute
+  def update_user_grant_creator_attribute
     case status
     when 'approved'
       requester.update_attribute(:grant_creator, true)
-      GrantCreatorRequestReviewMailer.with(request: self).approved.deliver_now
     when 'rejected'
```
Comment (fanglinnw) [inline app/models/grant_creator_request.rb]: seems redundant

Comment (jseraf) [reply app/models/grant_creator_request.rb]: Requester's :grant_creator boolean is nil or false. Once approved, the requester's grant_creator attribute needs to be set to true.  Should this be somewhere else?

File: spec/factories/grant_creator_request.rb (hunk 2 above)
Comment (jseraf) [reply spec/factories/grant_creator_request.rb]: typo

File: app/models/grant_creator_request.rb (hunk 4 above)
Comment (fanglinnw) [reply app/models/grant_creator_request.rb]: sorry i wasn't being clear. I meant the case switch has two branches doing the same thing: 
`requester.update_attribute(:grant_creator, false)`, in both `when 'rejected'` and `else`. these can be combined.

File: spec/models/grant_creator_request_spec.rb (hunk 3 above)
Comment (jseraf) [reply spec/models/grant_creator_request_spec.rb]: it did not

**Pull request merged**